### PR TITLE
 Fix always falling back to non-bgzip reader && allow .bgz extension

### PR DESCRIPTION
**Pull request by zamaudio**
Previously, files with .gz extension were assumed to be bgzip files.
Now, `.gz` is assumed gzip and `.bgz` is assumed bgzip.

I think it makes more sense this way, and hope the author agrees.

EDIT: This is not an adequate assumption, but I discovered that bgzip reader was never being used and is fixed below.

### CODE REVIEW

**Comment by brentp** [comment]
I think all that's required to get the desired effect is to change the single line:
```
	if len(config.Annotation) < runtime.GOMAXPROCS(0) && strings.HasSuffix(queryFile, ".gz") {	
```
to 
```
	if len(config.Annotation) < runtime.GOMAXPROCS(0) && (strings.HasSuffix(queryFile, ".gz") 
      || strings.HasSuffix(queryFile, ".bgz")) {	
```

this will make it so the query file can be .bgz, and [bix](https://github.com/brentp/bix) has been updated to support ".bgz" extension.

**Comment by zamaudio** [comment]
Hi @brentp, the line you posted above has slightly different behaviour than my PR, it will try to open both .gz and .bgz extensions as block gzipped input.  But my PR allows regular gzip to be opened as well.  With the current code, if you try to open a regular gzipped vcf file that has not been bgzipped, vcfanno crashes.
Ideally vcfanno would detect which kind of compression it is, but my go skills are not quite up to doing that.
Do you know of a way to detect if a file with .gz extension is block gzipped or regular gzipped?  I think that would solve the issue more robustly and not change existing behaviour of your program.

**Comment by brentp** [comment]
the current code allows regular gzip to be opened. (that's why it sets `qrdr = nil` when there's an error with bgzf reader). can you write a test that fails with master and passes with your code so I can understand what you are adding?

I still think that all that's needed is the line I posted above. If I misunderstand, then a test will help to clarify.

File: vcfanno.go [hunk 1]
```diff
@@ -128,29 +128,32 @@ see: https://github.com/brentp/vcfanno
...
 		if rdr, err := os.Open(queryFile); err == nil {
 			if st, err := rdr.Stat(); err == nil && st.Size() > 2320303098 {
 				qrdr, err = bgzf.NewReader(rdr, 4)
 				if err == nil {
-					log.Printf("using 4 worker threads to decompress query file")
+					log.Printf("using 4 worker threads to decompress bgzip file")
+				} else {
+					qrdr = nil
 				}
-				qrdr = nil
```
Comment (zamaudio) [inline vcfanno.go]: Without an else clause here, when bgzip reader succeeds OR fails, qrdr is unconditionally being set to nil, thus always falls back to non-bgzip reader (xopen)

**Comment by zamaudio** [comment]
@brentp I have identified that the bgzip reader is never being used in the current code.  I have provided a fix above that is basically your suggested change plus a couple of else clauses.

**Comment by brentp** [comment]
great! thank you for noticing and fixing this. I'll make a new release soon.

**Pull request merged**